Makefile

```diff
@@ -87,37 +81,35 @@ dev-reconcile-loop: build-dev ## Trigger the reconcile loop inside a container f
 		-e CONFIG=/work/config.dev.toml \
 		$(IMAGE_NAME)-dev:latest
 
-clean:
-	@rm -rf .tox .eggs reconcile.egg-info build .pytest_cache venv GIT_VERSION
+clean: ## Clean up the local development environment
+	@rm -rf .tox .eggs reconcile.egg-info build .pytest_cache venv .venv GIT_VERSION
 	@find . -name "__pycache__" -type d -print0 | xargs -0 rm -rf
 	@find . -name "*.pyc" -delete
 
 pypi-release:
-	@$(CONTAINER_ENGINE) build -t $(UUID):latest -f dockerfiles/Dockerfile.publish-release .
-	@$(CONTAINER_ENGINE) run -e TWINE_USERNAME -e TWINE_PASSWORD --rm $(UUID):latest ./build_tag.sh
-	@$(CONTAINER_ENGINE) rmi $(UUID):latest
+	@$(CONTAINER_ENGINE) build --progress=plain --build-arg TWINE_USERNAME --build-arg TWINE_PASSWORD --target pypi -f dockerfiles/Dockerfile .
+
+pypi:
+	uv build --sdist --wheel
+	uv publish
 
 dev-venv: clean ## Create a local venv for your IDE and remote debugging
-	python3.11 -m venv venv
-	@$(VENV_CMD) pip install --upgrade pip
-	@$(VENV_CMD) pip install -e .
-	@$(VENV_CMD) pip install -r requirements/requirements-dev.txt
+	uv sync --python 3.11
 
 print-files-modified-in-last-30-days:
 	@git log --since '$(shell date --date='-30 day' +"%m/%d/%y")' --until '$(shell date +"%m/%d/%y")' --oneline --name-only --pretty=format: | sort | uniq | grep -E '.py$$'
 
 format:
-	@$(VENV_CMD) ruff format
-	@$(VENV_CMD) ruff check
+	@uv run ruff format
+	@uv run ruff check
 
 gql-introspection:
 	# TODO: make url configurable
-	@$(VENV_CMD) qenerate introspection http://localhost:4000/graphql > reconcile/gql_definitions/introspection.json
+	@uv run qenerate introspection http://localhost:4000/graphql > reconcile/gql_definitions/introspection.json
 
 gql-query-classes:
-	@$(VENV_CMD) qenerate --version | grep -q $(EXPECTED_QENERATE_VERSION) || (echo "Bad qenerate version. Make sure you have version $(EXPECTED_QENERATE_VERSION) installed" && exit 1)
```

Comment:
Do we have an equivalent for this check? In the past we often saw confusion around non-matching qenerate versions. Or does `uv` automagically make sure that your local env is up-to-date whenever you run `@uv run ...`?

Reply:
Yes, `uv` takes care of it:

```
$ uv pip install qenerate==0.6.5
Resolved 7 packages in 516ms
Prepared 1 package in 135ms
Uninstalled 1 package in 3ms
Installed 1 package in 4ms
 - qenerate==0.7.0
 + qenerate==0.6.5

$ make gql-query-classes
   Built qontract-reconcile @file:///Users/cassing/workspace/github/chassing/qontract-reconcile
Uninstalled 2 packages in 2ms
Installed 2 packages in 3ms

$ uv pip freeze | grep qenerate                                                                                                                                               
qenerate==0.7.0
```